frontend/ui/widget/checkbutton.lua

```diff
@@ -67,11 +67,14 @@ function CheckButton:initCheckButton(checked)
             show_parent = self.show_parent or self,
         }
     end
+    local fgcolor = self.fgcolor or Blitbuffer.COLOR_BLACK
     self._textwidget = TextBoxWidget:new{
         text = self.text,
         face = self.face,
         width = (self.width or self.parent:getAddedWidgetAvailableWidth()) - self._checkmark.dimen.w,
-        fgcolor = self.enabled and Blitbuffer.COLOR_BLACK or Blitbuffer.COLOR_DARK_GRAY,
+        bold = self.bold,
+        fgcolor = self.enabled and fgcolor or Blitbuffer.COLOR_DARK_GRAY,
+        bgcolor = self.bgcolor,
```

Comment:
This will work if we provide GRAY colors?
I see you have kept those in - while you removed the TextBoxWidget addition that was using them.
(In spite of my diatribe about colours, I'm not opposed to keeping the TextBoxWidget color code even if we don't use it - there's been a few minutes of your brainwork put there, sad to lose it :) and user-patch-users may be able to being themselves colors into their world if that code is there.)

Reply:
This will work to provide selectable gray colors by setting `fgcolor=...`.
Color support requires the 24/32bit Blitbuffer type to be enabled in the widget (which I had ripped out for now).
I think bring back support in widgets can be left for a future project/MR to avoid feature creep.